presto-main/src/main/java/com/facebook/presto/sql/planner/PartitioningScheme.java

```diff
@@ -29,10 +29,40 @@
 
 public class PartitioningScheme
 {
+    public enum Replication
```

Comment:
why not also add method: `replicateNulls` instead of using `isReplicateNulls()`. This way it will be more objective and all `PartitioningScheme` have to declare how they replicate data instead of relying on `switch`.